### ISSUE TITLE
Repetitive calls of getApplicationsList when page reloads - getting menu options?

### ISSUE THREAD

**Issue by nmadruga** [openmsupply/conforma-web-app#1393]
@CarlosNZ as mentioned in the meeting on 17/10/2022 that `getApplicationsList` is a query that gets called on each page reload. I just observed the same phenomenon happening - this one is after I Apply to set each section with a Reviewer. The page is reloaded - since we call `fullStructure.reload()` in useUpdateAssignment hook (which I'm working on fixing now).

The thing is, any time we have a structure for the current application that causes the App to reload the page, and calls these fetches may times. I believe this is happening to rebuild the top menus, but will need further investigation.

You can use the same snapshot to test: `dev/snapshots/AA_Testing_Fully_reviewed_submission_3`. And load this Applications review page: http://localhost:3000/application/AIM-2022-0066/review?tab=assignment
Then add reviewers and click Apply - the page gets fully reloaded.

Menu has these 4 templates:
![Screen Shot 2022-10-18 at 3 11 14 PM](https://user-images.githubusercontent.com/16461988/196319722-4bb5a82f-bbcb-4329-a6fa-9dd00def4c7f.png)

Full lists of API Calls
![Screen Shot 2022-10-18 at 3 09 24 PM](https://user-images.githubusercontent.com/16461988/196319783-a0ecafd7-407b-4a07-bdf0-5aa247701a35.png)

Fetch
![Screen Shot 2022-10-18 at 3 11 04 PM](https://user-images.githubusercontent.com/16461988/196319768-7f127fa4-d148-4ce6-929b-7433bef4a090.png)
Result
![Screen Shot 2022-10-18 at 3 09 51 PM](https://user-images.githubusercontent.com/16461988/196319782-e6920394-9742-46cf-96e9-31b619a5f0bc.png)

Fetch
![Screen Shot 2022-10-18 at 3 10 49 PM](https://user-images.githubusercontent.com/16461988/196319770-00d61113-caac-49a5-862c-c36754036324.png)
Result
![Screen Shot 2022-10-18 at 3 10 08 PM](https://user-images.githubusercontent.com/16461988/196319779-bd11bc7d-794a-4f18-85e1-d1dc75fe84b9.png)

Fetch
![Screen Shot 2022-10-18 at 3 10 42 PM](https://user-images.githubusercontent.com/16461988/196319772-013d1f77-b220-4cf8-90bb-8768fa2bd241.png)
Result
![Screen Shot 2022-10-18 at 3 10 08 PM](https://user-images.githubusercontent.com/16461988/196319779-bd11bc7d-794a-4f18-85e1-d1dc75fe84b9.png)

Fetch
![Screen Shot 2022-10-18 at 3 10 34 PM](https://user-images.githubusercontent.com/16461988/196319773-6a656836-01b2-4552-bfa3-97c3d2c63306.png)
Result
![Screen Shot 2022-10-18 at 3 10 25 PM](https://user-images.githubusercontent.com/16461988/196319775-72a2b7c8-1d65-45b9-ae67-650fead80669.png)

**Comment by CarlosNZ** [comment]
Okay, thanks for that, I'll look into it. I think the top menu state gets reloaded when "templatePermissions" changes, but maybe we can tighten that up, or be more precise about what need to be monitored. (Or if templatePermissions is not actually changing, then that's kind of a bug)

**Comment by CarlosNZ** [comment]
> I believe this is happening to rebuild the top menus, but will need further investigation.

I don't think this has anything to do with the top menus. When I add console.logs to the "UserArea" component, I'm not seeing any rendering that I shouldn't. There's no updating going on when I change assignment in that snapshot you suggested.

> The thing is, any time we have a structure for the current application that causes the App to reload the page, and calls these fetches may times.

Yes there is, I suspect it's due to badly implemented `useEffect`s that respond to changed references to a variable without actually changing the content, something like I found was happening [here](https://github.com/openmsupply/conforma-web-app/pull/1364/files#r985097835).

> The page is reloaded - since we call fullStructure.reload() in useUpdateAssignment hook (which I'm working on fixing now).

What exactly needs fixing here? (FYI the *page* is not reloaded, the application is).

**Comment by nmadruga** [comment]
> What exactly needs fixing here? (FYI the _page_ is not reloaded, the application is).

Exactly. That's the problem I noticed. I think I have partially fixed in PR #1396, but I posted this here since you wasn't aware of how many calls to `getApplicationList` would be called when this page reload.

**Comment by nmadruga** [comment]
> What exactly needs fixing here?

The many calls to `getApplicationList` need to be investigated - if that's still happening after the PR mentioned has been reviewed. Maybe it was just part of the problem that was happening.
